src/Temperature-Accretech_UFseries/main.py

```diff
@@ -0,0 +1,160 @@
+except ModuleNotFoundError:
+    message_box = print
+    print(
+        "Accretech driver: Please use the latest version of SweepMe! to use this "
+        "driver to display a message box. "
+        "Fallback to 'print' method.",
+    )
+
+
+class Device(EmptyDevice):
+    def __init__(self):
+        EmptyDevice.__init__(self)
+
+        self.instance_key = None
+        self.port_str = None
+        self.prober: accretech_uf.AccretechProber = None
+        self.temperature: float = 0.0
+
+        self.shortname = "Accretech"
+        self.variables = ["Temperature"]
+        self.units = [""]  # unit of X and Y is either µm or inch, depending on system settings
+        self.plottype = [True]
+        self.savetype = [True]
+
+        self.port_manager = True
+        self.port_types = ["GPIB"]
+        self.port_properties = {
+            "timeout": 5,  # only for write&read messages, SRQ&STB commands have own timeout
+            "GPIB_EOLwrite": "\r\n",
+            "GPIB_EOLread": "\r\n",
+        }
+
+        self.verbosemode = True
+
+    def set_GUIparameter(self) -> dict:  # noqa: N802
+        """Set initial GUI parameters."""
+        return {
+            "SweepMode": ["None", "Temperature"],
+            "Temperature unit": ["°C"],
+        }
+
+    def get_GUIparameter(self, parameter: dict) -> None:  # noqa: N802
+        """Handle SweepMe GUI parameters."""
+        self.units[0] = parameter["Temperature unit"]
+
+        self.port_str = parameter["Port"]
+
+    def connect(self) -> None:
+        """Connect to the device and handle multiple instances."""
+        # creating an AccretechProber instance that handles all communication
+        # TODO: Handle multiple instances
+        self.instance_key = f"Accretech_UF_{self.port_str}"
+
+        if self.instance_key in self.device_communication:
+            self.port = self.device_communication[self.instance_key]
+        else:
+            self.device_communication[self.instance_key] = self.port
+
+        self.prober = accretech_uf.AccretechProber(self.port)
+
+        self.prober.set_verbose(self.verbosemode)
+
+    def disconnect(self) -> None:
+        """Disconnect from the device."""
+        del self.prober  # this makes sure the event mechanism is disabled
+
+    def initialize(self) -> None:
+        """Initialize the device."""
+        # TODO: Check
+        self.prober.reset_alarm()
+
+    def configure(self):
+        pass
+
+    def unconfigure(self):
+        pass
+
+    def apply(self) -> None:
+        """Apply the given temperature."""
+        temperature = self.value
+
+        max_temperature = 200.0
+        min_temperature = -55.0
+        if temperature > max_temperature or temperature < min_temperature:
+            msg = f"Accretech UF series: Temperature must be between {min_temperature} and {max_temperature} °C."
+            raise Exception(msg)
+
+        status = self.prober.set_chuck_temperature(temperature)
+        successful_status = 93
```

Comment:
Not fully sure yet, but probably it would be better to directly throw an error in accretech_uf.py if set_chuck_temperature fails. In this case the user needs to do some steps and the measurement should stop.